### PR TITLE
Feature/465 - Adds API module back into Masonite

### PR DESCRIPTION
**Pull request by josephmancuso**

### CODE REVIEW

File: src/masonite/commands/MakeControllerCommand.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def handle(self):
         content = render_stub_file(stub_path, name)
 
         filename = f"{name}.py"
-        path = controllers_path(filename)
+        path = config_path(filename)
```
Comment (girardinsamuel) [inline src/masonite/commands/MakeControllerCommand.py]: Why did you rename this 🤔 ?

Comment (josephmancuso) [reply src/masonite/commands/MakeControllerCommand.py]: was a mistake lol

File: src/masonite/api/Api.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return self
+
+    def generate_token(self):
+        secret = self.config.get("jwt").get("secret")
+        algorithm = self.config.get("jwt").get("algorithm")
+        expire_minutes = self.config.get("jwt").get("expires")
+        version = self.config.get("jwt").get("version")
+        if expire_minutes:
+            expire_minutes = (
+                pendulum.now(tz="GMT").add(minutes=expire_minutes).to_datetime_string()
```
Comment (girardinsamuel) [inline src/masonite/api/Api.py]: Should we use `GMT` or `UTC`here ?

File: src/masonite/api/commands/APIInstallCommand.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    def __init__(self, application):
+        super().__init__()
+        self.app = application
+
+    def handle(self):
+        stub_path = self.get_stub_config_path()
+
+        content = render_stub_file(stub_path, "api.py")
+        print(content)
```
Comment (girardinsamuel) [inline src/masonite/api/commands/APIInstallCommand.py]: a little print to remove..

File: src/masonite/api/commands/APIInstallCommand.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        path = config_path("api.py")
```
Comment (girardinsamuel) [inline src/masonite/api/commands/APIInstallCommand.py:28]: This block looks duplicated !

File: tests/features/api/test_api.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from tests import TestCase
+from src.masonite.api.facades import Api
+from src.masonite.headers import Header
+
+
+class TestApiModule(TestCase):
+    def test_can_can_token_from_input(self):
```
Comment (girardinsamuel) [inline tests/features/api/test_api.py]:
```suggestion
    def test_can_get_token_from_input(self):
```

File: tests/features/api/test_api.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        request = self.make_request()
+        request.input_bag.add_post("token", "secret")
+
+        self.assertEqual(Api.get_token(), "secret")
+
+    def test_can_can_token_from_input_from_header(self):
```
Comment (girardinsamuel) [inline tests/features/api/test_api.py]:
```suggestion
    def test_can_get_token_from_header(self):
```

File: src/masonite/input/InputBag.py [hunk 7]
```diff
@@ -187,3 +187,7 @@ def parse_dict(self, dictionary):
                 new_dict.update({name: value})
 
         return new_dict
+
+    def add_post(self, key, value):
```
Comment (girardinsamuel) [inline src/masonite/input/InputBag.py:191]: Where this method is needed ?

**Review by girardinsamuel — CHANGES_REQUESTED** [review]
Here is a quick review !

File: src/masonite/api/Api.py (hunk 2 above)
Comment (josephmancuso) [reply src/masonite/api/Api.py]: There is actually no difference. but to make it consistent ill change it to UTC. We use UTC in the masonite-orm repo for example. Good catch

File: src/masonite/api/commands/APIInstallCommand.py (hunk 3 above)
Comment (josephmancuso) [reply src/masonite/api/commands/APIInstallCommand.py]: good catch

File: src/masonite/api/commands/APIInstallCommand.py (hunk 4 above)
Comment (josephmancuso) [reply src/masonite/api/commands/APIInstallCommand.py:28]: good catch

File: src/masonite/input/InputBag.py (hunk 7 above)
Comment (josephmancuso) [reply src/masonite/input/InputBag.py:191]: I believe I use it in one of the tests

File: src/masonite/api/Api.py (hunk 2 above)
Comment (girardinsamuel) [reply src/masonite/api/Api.py]: oh okay ! I was thinking it was different I'll go check those terms then 😅

File: src/masonite/input/InputBag.py (hunk 7 above)
Comment (girardinsamuel) [reply src/masonite/input/InputBag.py:191]: oh okay yes makes sense

File: src/masonite/api/Api.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        if not expires:
+            return True
+
+        expired = pendulum.parse(expires, tz="UTC").is_past()
+
+        if expired:
+            return False
+
+        if version:
+            if unencrypted_token["version"] != version:
```
Comment (josephmancuso) [inline src/masonite/api/Api.py:60]: I just realized this check has to happen first or else it will never get here if the expiration is set to None. even if tokens never expire we still need to verify token version

**Comment by josephmancuso** [comment]
Fixed JWT Guard

**Comment by josephmancuso** [comment]
Also fixed a big issue where routes weren't even aloud to have multiple middleware stacks assigned to them

**Review by josephmancuso — COMMENTED** [review]
reviewed myself. Why can't we approve our own PR's?
